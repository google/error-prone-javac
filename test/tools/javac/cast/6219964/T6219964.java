/*
 * @test    /nodynamiccopyright/
 * @bug     6219964
 * @summary Compiler allows illegal cast of anonymous inner class
 * @compile/fail/ref=T6219964.out -XDrawDiagnostics  T6219964.java
 */

public class T6219964 {
    interface I { }
    void foo() {
        new Object() {
            I bar() {
                return (I)this;
            }
        };
    }
}
