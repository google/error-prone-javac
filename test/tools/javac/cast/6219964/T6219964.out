T6219964.java:13:27: compiler.err.prob.found.req: (compiler.misc.inconvertible.types: compiler.misc.anonymous.class: java.lang.Object, T6219964.I)
1 error
